### DISCUSSION TITLE
Class Variable Support

### DISCUSSION THREAD

**Discussion by esb** [ruby2js/ruby2js#209] [Q&A]
From my understanding, ```@@``` class variables are a little bit restricted.

I'm not interested in private variables, so I use ```underscored_private: true``` with ES2022.

If I define ```@@variable = nil``` in ```MyClass```, I get ```MyClass._variable = null```

It would be nice to have ```this.constructor._variable``` generated instead, but I guess that's not important.

The real problem for me is that I'm interfacing with base and sub-classes that are native JS and they use specific static variable names that I can't control. For example, in this instance, the static variable would be defined as ```static variable = null```, so I've got an obvious mismatch as ruby2js will generate the name with a ```_``` prefix.

Is there anyway that this could be configured?

**Comment by jaredcwhite** [comment]
@esb Yeah I like that idea, started noodling around with an update. What's a good config option I wonder? `public_class_variables`?

**Comment by esb** [comment]
Would it be useful to allow the prefix to be defined? The default would be "_", but it could be set to "" or false to disable the prefix entirely.
